src/main/java/org/mozilla/magnet/scanner/MagnetScanner.java

```diff
@@ -31,11 +46,23 @@
  *
  * @author Francisco Jordano
  */
-public class MagnetScanner {
+public class MagnetScanner implements GoogleApiClient.ConnectionCallbacks, GoogleApiClient.OnConnectionFailedListener, MagnetScannerListener {
```

Comment:
Is a bit weird to me this class implements MagnetScannerListener, when it's aggregating different scanners.